src/Product/WCProductAdapter.php

```diff
@@ -1017,13 +1017,20 @@ protected function rule_match_conditions( array $rule ): bool {
 	 */
 	protected function get_product_taxonomy( $taxonomy ) {
 		$product = $this->get_wc_product();
-		$values  = get_the_terms( $product->get_id(), $taxonomy );
 
-		if ( ! $values ) {
+		if ( $product->is_type( 'variation' ) ) {
+			$values = $product->get_attribute( $taxonomy );
+			$values = explode( ', ', $values );
```

Comment:
I think I was partially wrong in my [previous comment](https://github.com/woocommerce/google-listings-and-ads/pull/1927#discussion_r1152837087). The product type `variation` has a different [`get_attribute` function](https://github.com/woocommerce/woocommerce/blob/7.6.1/plugins/woocommerce/includes/class-wc-product-variation.php#L135), which only returns a singular value, so the explode here isn't useful.

Reply:
After changes in https://github.com/woocommerce/google-listings-and-ads/commit/fe3ebb26a5f6baabdedc4dee8a15fd0e94a33551  this is not applicable anymore.